neurolib/models/subdivwc/timeIntegration.py

```diff
@@ -0,0 +1,293 @@
+        w2,
+        w3,
+        w4,
+        w5,
+        w6,
+        w7,
+        alpha_exc,
+        alpha_inh_s,
+        alpha_inh_d,
+        theta_exc,
+        theta_inh_s,
+        theta_inh_d,
+        q,
+        noise_exc,
+        noise_inh_s,
+        noise_inh_d,
+        exc_ou,
+        inh_s_ou,
+        inh_d_ou,
+        exc_ou_mean,
+        inh_s_ou_mean,
+        inh_d_ou_mean,
+        tau_ou,
+        sigma_ou,
+    )
+
+
+@numba.njit
+def timeIntegration_njit_elementwise(
+        startind,
+        t,
+        dt,
+        sqrt_dt,
+        N,
+        Cmat,
+        K_gl,
+        Dmat_ndt,
+        excs,
+        inh_ss,
+        inh_ds,
+        exc_input_d,
+        inh_s_input_d,
+        inh_d_input_d,
+        exc_ext,
+        inh_s_ext,
+        inh_d_ext,
+        tau_exc,
+        tau_inh_s,
+        tau_inh_d,
+        w1,
+        w2,
+        w3,
+        w4,
+        w5,
+        w6,
+        w7,
+        alpha_exc,
+        alpha_inh_s,
+        alpha_inh_d,
+        theta_exc,
+        theta_inh_s,
+        theta_inh_d,
+        q,
+        noise_exc,
+        noise_inh_s,
+        noise_inh_d,
+        exc_ou,
+        inh_s_ou,
+        inh_d_ou,
+        exc_ou_mean,
+        inh_s_ou_mean,
+        inh_d_ou_mean,
+        tau_ou,
+        sigma_ou,
+):
+    ### integrate ODE system:
+
+    def F_exc(x, Theta, A):
+        return (alpha_exc/(alpha_exc+q*A))*((1/(1+np.exp(-alpha_exc*(x-(theta_exc+Theta+(1-q)*A)))))
+                                            -(1/(1+np.exp(alpha_exc*theta_exc))))
+
+    def F_inh_s(x, Theta, A):
+        return (alpha_inh_s/(alpha_inh_s+q*A))*((1/(1+np.exp(-alpha_inh_s*(x-(theta_inh_s+Theta+(1-q)*A)))))
+                                            -(1/(1+np.exp(alpha_inh_s*theta_inh_s))))
+    def F_inh_d(x, Theta, A):
+        return (alpha_inh_d/(alpha_inh_d+q*A))*((1/(1+np.exp(-alpha_inh_d*(x-(theta_inh_d+Theta+(1-q)*A)))))
+                                            -(1/(1+np.exp(alpha_inh_d*theta_inh_d))))
+
+    k_exc = np.exp(alpha_exc)/(1+np.exp(alpha_exc))
+    k_inh_s = np.exp(alpha_inh_s)/(1+np.exp(alpha_inh_s))
+    k_inh_d = np.exp(alpha_inh_d)/(1+np.exp(alpha_inh_d))
+
+    for i in range(startind, startind + len(t)):
+
+        # loop through all the nodes
+        for no in range(N):
+
+            # To save memory, noise is saved in the activity array
+            noise_exc[no] = excs[no, i]
+            noise_inh_s[no] = inh_ss[no, i]
+            noise_inh_d[no] = inh_ds[no, i]
+
+            # delayed input to each node
+            exc_input_d[no] = 0
+            inh_s_input_d[no] = 0
+            inh_d_input_d[no] = 0
+
+            for l in range(N):
+                exc_input_d[no] += K_gl * Cmat[no, l] * (excs[l, i - Dmat_ndt[no, l] - 1])
+
+
+
+            # subtractive/divisive Wilson-Cowan model
+            exc_rhs = (
+                1/tau_exc*(-excs[no, i - 1]+(k_exc-excs[no, i - 1])*F_exc(w1*excs[no, i - 1]+exc_ext[no],
```

Comment:
```suggestion
                1/tau_exc*(-excs[no, i - 1]+(k_exc-excs[no, i - 1])*F_exc(w1*excs[no, i - 1]+exc_ext[no] + exc_input_d ,
```
maybe `+ exc_input_d`? otherwise I do not see any use of network input `exc_input_d ` :)

Reply:
by the way I went and check basic 2population WC model already in `neurolib` and I think I found a bug.
In the original WC model (`neurolib/models/wc/timeIntegration.py`) both excitatory and inhibitory population got ` + exc_input_d` into sigmoid transfer function and I feel from the WC definition, only excitatory should get network input from other excitatory nodes

Reply:
> by the way I went and check basic 2population WC model already in `neurolib` and I think I found a bug.
> In the original WC model (`neurolib/models/wc/timeIntegration.py`) both excitatory and inhibitory population got ` + exc_input_d` into sigmoid transfer function and I feel from the WC definition, only excitatory should get network input from other excitatory nodes

Thank for finding the bug! We need to fix it!

Reply:
not sure if it's a bug, I worked with WC model few years back, so it's better if @ChristophMetzner confirms :)

Reply:
Yes, we should wait for @ChristophMetzner. However, every implementation of a brain network that I've seen so far only uses excitatory coupling.

Reply:
Thanks for finding that bug/those bugs. You're right, it should be `+ exc_input_d`.
Wrt to the original WC model, I think that you're right that most (all?) people only give input from other network nodes to the excitatory population and we should stick to that.

Reply:
I will fix those bugs asap (probably tomorrow).